docs/2.9.0/docs/canton/usermanual/performance.rst

```diff
@@ -506,74 +506,104 @@ If you submit a ledger API command (which can have multiple Daml commands), then
 
-For example, if the below `doUpdate` is only visible to the `owner` party, then whether the `efficient` flag is set or not 
+In general, if you write batch commands, group the operations on the ledger based on the informees' participants and create
+a single choice and batch operation per informee's participants group.
+
+For example, if the `doUpdate` is only visible to the `owner` party, then whether the `efficient` flag is set or not
 will have a huge impact on the number of views created.
 
 .. code:: daml
 
    doUpdate : Bool -> Party -> Party -> [ContractId Example] -> Update ()
-   doUpdate efficient owner obs batch = do 
-   if efficient then do 
+   doUpdate efficient owner obs batch = do
+   if efficient then do
       -- Would be even more efficient if we can reuse the BatchFoo contract
-      batcher <- create BatchFoo with owner 
-      batcher `exercise` Run with batch = batch; obs = obs    
+      batcher <- create BatchFoo with owner
+      batcher `exercise` Run with batch = batch; obs = obs
    else do
       -- Canton will create one view per exercise
-      forA_ batch (`exercise` Foo)    
+      forA_ batch (`exercise` Foo)
    pure ()
 
    template Example
-   with 
-      owner : Party 
-      obs: Party 
-   where 
-      signatory owner 
-      observer obs 
-
+   with
+      owner : Party
+      obs: Party
+   where
+      signatory owner
+      observer obs
+
+      -- even though the participants to be informed are a subset of the contract's informee participants
+      -- they don't get merged into a single view if this choice is called independently each time (multiple root actions)
       choice Foo : ()
-         controller owner 
-         do 
+         controller owner
+         do
          return ()
 
    template BatchFoo
-   with 
-      owner : Party 
-   where 
-      signatory owner 
+   with
+      owner : Party
+   where
+      signatory owner
 
       choice Run : ()
-         with  
+         with
          batch : [ContractId Example]
          obs : Party
          -- The observer here is a choice observer. Therefore, Canton will
-         -- ship a single view with the top node of this choice if the observer of the 
-         -- choice aligns with the observer of the contract.       
-         observer obs
-         controller owner 
+         -- ship a single view with the top node of this choice if the observer of the
+         -- choice aligns with the observer of the contract. In other words, either they are the same or the new observer
+         -- is hosted by any of the contract's informee participants.
+         observer new_obs
+         controller owner
          do
          forA_ batch (`exercise` Foo)
 
-
-As a rule, the number of views should depend on the number of stakeholder groups you have in your batch choice, 
-not the number of "batches" you process in parallel within one command.
+As a rule, the number of views should depend on the number of participant stakeholder groups
```

Comment:
What is a participant stakeholder group and how does it differ from the concept of informee participants?

Reply:
I was using them as synonyms, but in this phrase it probably creates confusion since we use _"informee participants"_ right after.